src/contexts/SiteLaunchContext.tsx

```diff
@@ -29,6 +31,8 @@ interface SiteLaunchContextProps {
   increasePageNumber: () => void
   decreasePageNumber: () => void
   pageNumber: SiteLaunchPageIndex
+  handleIncrementStepNumber: () => void
+  handleDecrementStepNumber: () => void
```

Comment:
we need to take care to prevent props explosion here. this is not an issue for now because idt we will have active work going on for site launch, but we want to avoid the case where we just put everything here and ignore proper component structure, which will impact maintainability in the future.

tl;dr: is ok but we should avoid adding stuff here already.

Reply:
can, but for future ref may I trouble you to explain how I could make this more maintainable + allow storybook to be versatile enough to be able to click thr the checkboxes? I am abit unsure of how I could have made this PR better in the future

Reply:
without pair programming/actually working on this portion of the codebase, it's bit difficult to actually find something that works well. 1 sign that you could use to signal for help could be that you find yourself needing the context for everything, which shouldn't be the case.

1 thing that you can try is to see which parts of the state your child components truly need and refactoring so that you pass that state to your children. i've done htis [here](https://github.com/isomerpages/isomercms-frontend/pull/1431/commits/b82157bc034c48799679131bf1c76e94831ca7bc) because the two nested components were p disjoint in their props but only really needed the `selectedOption`